### DISCUSSION TITLE
enabling-prerendering-for-blazor-webassembly-apps/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#37] [Announcements]
# enabling-prerendering-for-blazor-webassembly-apps/

In this post I describe how to host a Blazor WebAssembly app in an ASP.NET Core app and how to enable WebAssembly prerendering

https://andrewlock.net/enabling-prerendering-for-blazor-webassembly-apps/

**Comment by rockxl1** [comment]
Very good post! Better than the official :)

Btw how to inject IJSRuntime, NavegationManager and other blazor interfaces in host project?

**Reply by devdestech** [reply]
Hello Andrew, Rockxl1... were you be able resolved?
I have a problem with prerendering anf IJSRuntime.

InvalidOperationException: JavaScript interop calls cannot be issued during server-side prerendering, because the page has not yet loaded in the browser. Prerendered components must wrap any JavaScript interop calls in conditional logic to ensure those interop calls are not attempted during prerendering.

[a comment by an automated account is not shown]

**Comment by ckirker** [comment]
If using .NET6 and using **top-level statements** the final structure of the server Program.cs is as follows:

``` C#
var builder = WebApplication.CreateBuilder(args);

builder.Services.AddRazorPages();

var app = builder.Build();

if (app.Environment.IsDevelopment()) {
    app.UseDeveloperExceptionPage();
    app.UseWebAssemblyDebugging();
} else {
    app.UseStatusCodePages();
    app.UseHsts();
}

app.UseHttpsRedirection();
app.UseBlazorFrameworkFiles();
app.UseStaticFiles();

app.UseRouting();

app.MapRazorPages();

app.MapFallbackToPage("/_Host");

app.Run();
```

Hope this helps.
